src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -1994,6 +2057,123 @@ public void ChainTHasCheckpointAt1000_MaxReorgIs10_TwoChainsPriorTo1000Presented
             consumedHeaders = connectionResult.Consumed.ToArray(listOfChainAHeaders.Count);
             consumedHeaders.HaveBlockDataAvailabilityStateOf(BlockDataAvailabilityState.HeaderOnly).Should().BeTrue();
         }
+      
+        /// <summary>
+        /// Issue 19 @ New peer K claims 10d. Peer K disconnects. Chain shouldn't change.
+        /// </summary>
+        [Fact]
+        public void NewPeerClaimsHeadSecondTime_PeerDisconnected_ChainShouldNotChange()
+        {
+            const int initialChainSize = 5;
+            TestContext ctx = new TestContextBuilder().WithInitialChain(initialChainSize).UseCheckpoints().Build();
+            ChainedHeaderTree cht = ctx.ChainedHeaderTree;
+            ChainedHeader initialChainTip = ctx.InitialChainTip;
+
+            IEnumerable<BlockHeader> listOfAllUniqueBlockHeaders = ctx.SetupPeersForTest(initialChainSize, ctx, cht, initialChainTip, out ChainedHeader chainATip, out ChainedHeader chainDTip);
+            CheckChainedHeaderTreeConsistency(cht, listOfAllUniqueBlockHeaders);
+
+            // Additional SetUp for current test.
+            ChainedHeader chainKTip = ctx.ExtendAChain(0,chainDTip); //peer K has exactly the same chain as peer D.
```

Comment:
Why not just use the same header? `ChainedHeader chainKTip = chainDTip;`

ExtendAChain with first argument 0 will return you the input parameter which is chainDTip in current case. So there is no need to add this complication.